src/analysis/rust_type.rs

```diff
@@ -53,21 +87,36 @@ fn rust_type_full(env: &Env, type_id: library::TypeId, nullable: Nullable, ref_m
 
                 UniChar => ok("char"),
                 Utf8 => if ref_mode.is_ref() { ok("str") } else { ok("String") },
-                Filename => if ref_mode.is_ref() { ok("std::path::Path") } else { ok("std::path::PathBuf") },
-
+                Filename => {
+                    if ref_mode.is_ref() {
+                        ok("std::path::Path")
+                    }
+                    else {
+                        ok("std::path::PathBuf")
+                    }
+                }
                 Type => ok("glib::types::Type"),
                 Unsupported => err("Unsupported"),
                 _ => err(&format!("Fundamental: {:?}", fund)),
             }
         },
-        Alias(ref alias) => rust_type_full(env, alias.typ, nullable, ref_mode)
-                .map_any(|_| alias.name.clone()),
-
-        Enumeration(ref enum_) => Ok(enum_.name.clone()),
-        Bitfield(ref bitfield) => Ok(bitfield.name.clone()),
-        Record(ref record) => Ok(record.name.clone()),
-        Interface(ref interface) => Ok(interface.name.clone()),
-        Class(ref klass) => Ok(klass.name.clone()),
+        Alias(ref alias) => {
+            rust_type_full(env, alias.typ, nullable, ref_mode)
+                .map_any(|_| alias.name.clone())
```

Comment:
For something like `Rectangle` alias for `RectangleInt`. (Just theoretical example)